### PR TITLE
feat: Add custom suggestion as a prop

### PR DESCRIPTION
**Pull request by andresmechali**
This feature allows to pass a default functional component as a prop and override default suggestion component. I wrote it because I need it for my own project, in order to show different icons based on the type of suggestion. I would be glad if you reviewed it and considered adding this feature to your library so that anyone can use it. I would also be glad to receive some feedback, as I have no experience contributing to open source libraries.
Thanks!

### CODE REVIEW

**Comment by andresmechali** [comment]
anyone reviewing please?

File: README.md [hunk 1]
```diff
@@ -157,6 +157,7 @@ Option | Type | Default | Description
 |[`inline`](#inline) | `Boolean` | `true` | Render input field and selected tags in-line
 |[`allowUnique`](#allowUnique) | `Boolean` | `true` | Boolean value to control whether tags should be unqiue
 |[`allowDragDrop`](#allowDragDrop) | `Boolean` | `true` | Boolean value to control whether tags should have drag-n-drop features enabled
+|[`customSuggestion`](#customSuggestion) | `Function` | `undefined` | Function that returns a custom suggestion for overriding the default one
```
Comment (ad1992) [inline README.md]: can you rename this to `renderSuggestion`

File: README.md [hunk 2]
```diff
@@ -441,6 +442,16 @@ This prop controls whether tags should be unique.
 #### allowDragDrop (optional, defaults to `true`)
 This prop controls whether tags should have the drag-n-drop feature enabled.
 
+<a name="customSuggestion"></a>
+#### customSuggestion (optional)
+This props allows to override the default suggestion component and return a custom one. It receives the suggestion and the query string as parameters. For example:
```
Comment (ad1992) [inline README.md]: `Provide your own suggestion renderer `

File: src/components/ReactTags.js [hunk 3]
```diff
@@ -416,6 +417,7 @@ class ReactTags extends Component {
           shouldRenderSuggestions={this.props.shouldRenderSuggestions}
           isFocused={this.state.isFocused}
           classNames={this.state.classNames}
+          customSuggestion={this.props.customSuggestion}
```
Comment (ad1992) [inline src/components/ReactTags.js]: `renderSuggestion`

File: __tests__/suggestions.test.js [hunk 4]
```diff
@@ -206,4 +206,19 @@ describe('Suggestions', function() {
...
+        renderSuggestion: ({ text }) => (
+          <div>
+            <i />
+            {text}
+          </div>
+        ),
+      })
+    );
+
+    expect($el.find('i').length).to.equal(4);
```
Comment (ad1992) [inline __tests__/suggestions.test.js]: instead checking `i` can you add a class to the div in line 214 and check then assert with classname here

File: README.md [hunk 5]
```diff
@@ -157,6 +157,7 @@ Option | Type | Default | Description
 |[`inline`](#inline) | `Boolean` | `true` | Render input field and selected tags in-line
 |[`allowUnique`](#allowUnique) | `Boolean` | `true` | Boolean value to control whether tags should be unqiue
 |[`allowDragDrop`](#allowDragDrop) | `Boolean` | `true` | Boolean value to control whether tags should have drag-n-drop features enabled
+|[`renderSuggestion`](#renderSuggestion) | `Function` | `undefined` | Function that returns a custom suggestion for overriding the default one
```
Comment (ad1992) [inline README.md]: `Render prop for rendering your own suggestions`

File: src/components/Suggestions.js [hunk 6]
```diff
@@ -84,8 +85,16 @@ class Suggestions extends Component {
     return query.length >= minQueryLength && isFocused;
   };
 
+  renderSuggestion = (item, query) => {
+    if (this.props.renderSuggestion) {
```
Comment (ad1992) [inline src/components/Suggestions.js]: can you destruct `renderSuggestion` and check here `typeof renderSuggestion === 'function'`?

File: __tests__/suggestions.test.js (hunk 4 above)
Comment (andresmechali) [reply __tests__/suggestions.test.js]: done

File: README.md (hunk 5 above)
Comment (andresmechali) [reply README.md]: done

File: src/components/Suggestions.js (hunk 6 above)
Comment (andresmechali) [reply src/components/Suggestions.js]: done

**Comment by andresmechali** [comment]
@ad1992 just finished making these changes. Thanks for your feedback!

File: __tests__/suggestions.test.js [hunk 7]
```diff
@@ -206,4 +206,19 @@ describe('Suggestions', function() {
 
     expect($el.componentDidUpdate.called).to.equal(true);
   });
+
+  test('should render custom suggestions', function() {
```
Comment (ad1992) [inline __tests__/suggestions.test.js]: `should render custom suggestions when renderSuggestion prop is provided`

Comment (andresmechali) [reply __tests__/suggestions.test.js]: done

**Comment by ad1992** [comment]
@andresmechali thanks for your contribution.

**Pull request merged**